### PR TITLE
Tracking OPC fees

### LINKED ISSUE #589: FixedRateExchange:  add OPC fees to fixedRateExchangeSwap entity

### PR DESCRIPTION
**Pull request by jamiehewitt15**
Fixes #589 

Changes proposed in this PR:

- Adding `oceanFeeAmount` to `fixedRateExchangeSwap` entity 
- Updating tests

### CODE REVIEW

File: src/mappings/fixedRateExchange.ts [hunk 1]
```diff
@@ -176,6 +176,12 @@ export function handleSwap(event: Swapped): void {
   swap.exchangeId = fixedRateId
   swap.by = getUser(event.params.by.toHex()).id
 
+  // Track OPC swap fee
+  swap.oceanFeeAmount = weiToDecimal(
+    event.params.oceanFeeAmount.toBigDecimal(),
+    BigInt.fromI32(18).toI32()
```
Comment (alexcos20) [inline src/mappings/fixedRateExchange.ts:182]: OPF Fee is always taken in baseToken, so it's not always using 18 decimals.
You should use something like:
```
swap.oceanFeeAmount= weiToDecimal(
    event.params.oceanFeeAmount.toBigDecimal(),
    BigInt.fromI32(baseToken.decimals).toI32()
  )
  ```
  and move this code after we have baseToken  (like before swap.save() )

**Review by alexcos20 — CHANGES_REQUESTED** [review]

**Review by alexcos20 — CHANGES_REQUESTED** [review]

File: test/integration/FixedRateExchange.test.ts [hunk 2]
```diff
@@ -648,6 +650,7 @@ describe('Fixed Rate Exchange tests', async () => {
     assert(swaps.block === tx.blockNumber, 'incorrect value for: block')
     assert(swaps.createdTimestamp >= time, 'incorrect: createdTimestamp')
     assert(swaps.createdTimestamp < time + 25, 'incorrect: createdTimestamp 2')
+    assert(swaps.oceanFeeAmount === oceanFeeAmount, 'incorrect: oceanFeeAmount')
```
[the remark this answers is not in the document]
Comment (jamiehewitt15) [reply test/integration/FixedRateExchange.test.ts:657]: Yeah good point. I've updated this, thanks for the feedback

**Review by alexcos20 — APPROVED** [review]

**Comment by alexcos20** [comment]
I know that original issue was about OPFC fees, but let's add marketFeeAmount and consumeMarketFeeAmount as well

**Comment by jamiehewitt15** [comment]
> I know that original issue was about OPFC fees, but let's add marketFeeAmount and consumeMarketFeeAmount as well

@alexcos20 I've added them, can you let me know if you're happy with the changes?

**Comment by alexcos20** [comment]
lgtm, we can merge it
Thanks

**Pull request merged**